lxd/groups.go

```diff
@@ -0,0 +1,1361 @@
+			return err
+		}
+
+		apiGroup, err := group.ToAPI(ctx, tx.Tx())
+		if err != nil {
+			return err
+		}
+
+		err = util.EtagCheck(r, *apiGroup)
+		if err != nil {
+			return err
+		}
+
+		if groupPut.Description != "" {
+			err = cluster.UpdateGroup(ctx, tx.Tx(), groupName, cluster.Group{
+				Name:        groupName,
+				Description: groupPut.Description,
+			})
+			if err != nil {
+				return err
+			}
+		}
+
+		newPermissions := make([]api.Permission, 0, len(groupPut.Permissions))
+		for _, permission := range groupPut.Permissions {
+			if !shared.ValueInSlice(permission, apiGroup.Permissions) {
+				newPermissions = append(newPermissions, permission)
+			}
+		}
+
+		permissionIDs, err := upsertPermissions(ctx, tx.Tx(), newPermissions)
+		if err != nil {
+			return err
+		}
+
+		groupPermissions := make([]cluster.GroupPermission, 0, len(permissionIDs))
+		for _, permissionID := range permissionIDs {
+			groupPermissions = append(groupPermissions, cluster.GroupPermission{
+				GroupID:      group.ID,
+				PermissionID: permissionID,
+			})
+		}
+
+		err = cluster.CreateGroupPermissions(ctx, tx.Tx(), groupPermissions)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	return response.EmptySyncResponse
+}
+
+// swagger:operation POST /1.0/auth/groups/{groupName} groups group_post
+//
+//	Rename the group
+//
+//	Renames the group
+//
+//	---
+//	consumes:
+//	  - application/json
+//	produces:
+//	  - application/json
+//	parameters:
+//	  - in: body
+//	    name: group
+//	    description: Update request
+//	    schema:
+//	      $ref: "#/definitions/GroupPost"
+//	responses:
+//	  "200":
+//	    $ref: "#/responses/EmptySyncResponse"
+//	  "400":
+//	    $ref: "#/responses/BadRequest"
+//	  "403":
+//	    $ref: "#/responses/Forbidden"
+//	  "500":
+//	    $ref: "#/responses/InternalServerError"
+func renameGroup(d *Daemon, r *http.Request) response.Response {
+	groupName, err := url.PathUnescape(mux.Vars(r)["groupName"])
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	var groupPost api.GroupPost
+	err = json.NewDecoder(r.Body).Decode(&groupPost)
+	if err != nil {
+		return response.BadRequest(fmt.Errorf("Invalid request body: %w", err))
+	}
+
+	err = validateGroupName(groupPost.Name)
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	ctx, cancel := context.WithTimeout(r.Context(), 10*time.Second)
+	defer cancel()
+
+	s := d.State()
+	err = s.DB.Cluster.Transaction(ctx, func(ctx context.Context, tx *db.ClusterTx) error {
+		err = cluster.RenameGroup(ctx, tx.Tx(), groupName, groupPost.Name)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return response.SmartError(err)
+	}
+
+	// When a group is renamed with need to update list of groups names associated with each identity in the cache.
+	// (When a group is otherwise modified, the name is unchanged, so the cache doesn't need to be updated).
+	s.UpdateIdentityCache()
```

Comment:
How will this also be applied on other cluster members?

Reply:
Along with https://github.com/canonical/lxd/pull/12966 this is now addressed. Each handler which needs to update the identity cache notifies other cluster members of the change so that they also perform a cache refresh.